posthog/temporal/batch_exports/temporary_file.py

```diff
@@ -14,8 +14,25 @@
 import pyarrow.parquet as pq
 
 
+def replace_broken_unicode(obj):
+    if isinstance(obj, str):
+        return obj.encode("utf-8", "replace").decode("utf-8")
+    elif isinstance(obj, list):
+        return [replace_broken_unicode(item) for item in obj]
+    elif isinstance(obj, dict):
+        return {replace_broken_unicode(key): replace_broken_unicode(value) for key, value in obj.items()}
+    else:
+        return obj
+
+
 def json_dumps_bytes(d) -> bytes:
-    return orjson.dumps(d, default=str)
+    try:
+        return orjson.dumps(d, default=str)
+    except (UnicodeEncodeError, TypeError):
```

Comment:
Strange that the exception raised doesn't match the docs:
> If orjson.dumps() is given a str that does not contain valid UTF-8, orjson.JSONEncodeError is raised. If loads() receives invalid UTF-8, orjson.JSONDecodeError is raised.

However, I verified and it's indeed raising a `UnicodeEncodeError`. Wondering if we should be safe and catch `orjson.JSONEncodeError` just in case this is a bug that's fixed upstream at some point.

Reply:
Ok, I've figured it out: `orjson.JSONEncodeError` is an alias of `TypeError` (see: https://github.com/ijl/orjson/blob/master/src/typeref.rs#L197, and [this issue](https://www.github.com/ijl/orjson/issues/238))

Reply:
This means that this:
```
try:
    return orjson.dumps(d, default=str)
except orjson.JSONEncodeError:
    ...
```

Would also work (since `orjson.JSONEncodeError is TypeError`).

Reply:
One thing though: `UnicodeEncodeError` is not the exception being raised, but the `__cause__`, so I would consider dropping that from the `except` and keeping just `TypeError` or `orjson.JSONEncodeError`.